serial_driver/include/serial_driver/serial_driver.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 LeoDrive, Copyright 2021 The Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SERIAL_DRIVER__SERIAL_DRIVER_HPP_
+#define SERIAL_DRIVER__SERIAL_DRIVER_HPP_
+
+#include <memory>
+#include <string>
+
+#include "io_context/io_context.hpp"
+#include "serial_driver/serial_port.hpp"
+
+namespace drivers
+{
+namespace serial_driver
+{
+
+class SerialDriver
+{
+public:
+  explicit SerialDriver(const IoContext & ctx);
+
+  void init_port(const std::string & device_name, const SerialPortConfig & config);
```

Comment:
for the `udp_driver` we have these separated out into a `init_sender` and `init_receiver`. do serial ports by default have both built in? would there be any benefits to separating them out?

Reply:
The serial protocol, by default, can not connect a port without both send and receive. It's built in all the way down to the wiring (TX and RX wires). Since UDP is, by defnition, a one-way communication mechanism, splitting them makes sense. For serial, I would argue that it does not.